### PR TITLE
[MXNET-392] Support for a bunch of unary functions for csr matrices

### PR DESCRIPTION
**Pull request by haojin2**
## Description ##
As title

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Support for relu, sign, round, ceil, floor, trunc, fix, sqrt, cbrt, log1p, expm1, sin, tan, arcsin, arctan, sinh, arcsinh, tanh, arctanh, degrees, radians
- [x] Corresponding unit tests

### CODE REVIEW

**Comment by haojin2** [comment]
@eric-haibin-lin

File: src/operator/tensor/elemwise_unary_op_basic.cc [hunk 1]
```diff
@@ -70,7 +70,7 @@ static bool IdentityAttrLikeRhsStorageType(const nnvm::NodeAttrs& attrs,
 }
 
 // relu
-MXNET_OPERATOR_REGISTER_UNARY(relu)
+MXNET_OPERATOR_REGISTER_UNARY_WITH_RSP_CSR(relu, cpu, mshadow_op::relu)
```
Comment (eric-haibin-lin) [inline src/operator/tensor/elemwise_unary_op_basic.cc]: We probably want to move MXNET_ADD_SPARSE_OP_ALIAS inside MXNET_OPERATOR_REGISTER_UNARY_WITH_RSP_CSR. Do you mind changing that?

File: src/operator/tensor/elemwise_unary_op_basic.cc [hunk 2]
```diff
@@ -81,11 +81,9 @@ The storage type of ``relu`` output depends upon the input storage type:
 
    - relu(default) = default
    - relu(row_sparse) = row_sparse
+   - relu(csr) = csr
```
Comment (eric-haibin-lin) [inline src/operator/tensor/elemwise_unary_op_basic.cc]: Please add fluent methods to CSRNDArray documentation like what was did for RowSparseNDArray https://github.com/apache/incubator-mxnet/blob/master/docs/api/python/ndarray/sparse.md#trigonometric-functions 
And add tests in https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_sparse_ndarray.py#L847

File: src/operator/tensor/elemwise_unary_op_basic.cc (hunk 1 above)
Comment (haojin2) [reply src/operator/tensor/elemwise_unary_op_basic.cc]: Actually none of MXNET_OPERATOR_REGISTER_UNARY_WITH_RSP_CSR/MXNET_OPERATOR_REGISTER_UNARY_WITH_RSP/MXNET_OPERATOR_REGISTER_UNARY_WITH_SPARSE_DR has MXNET_ADD_SPARSE_OP_ALIAS in them, do you want to add that to all of them?

File: src/operator/tensor/elemwise_unary_op_basic.cc (hunk 2 above)
Comment (haojin2) [reply src/operator/tensor/elemwise_unary_op_basic.cc]: Done

File: src/operator/tensor/elemwise_unary_op_basic.cc (hunk 1 above)
Comment (eric-haibin-lin) [reply src/operator/tensor/elemwise_unary_op_basic.cc]: Add for `MXNET_OPERATOR_REGISTER_UNARY_WITH_RSP_CSR` and  `MXNET_OPERATOR_REGISTER_UNARY_WITH_RSP`. Lets skip the ones with SPARSE_DR

Comment (haojin2) [reply src/operator/tensor/elemwise_unary_op_basic.cc]: Done

File: docs/api/python/ndarray/sparse.md [hunk 3]
```diff
@@ -520,7 +583,7 @@ We summarize the interface for each class in the following sections.
 ```eval_rst
 
 .. autoclass:: mxnet.ndarray.sparse.CSRNDArray
-    :members: shape, context, dtype, stype, data, indices, indptr, copy, copyto, as_in_context, asscipy, asnumpy, asscalar, astype, tostype, slice, wait_to_read, zeros_like, __neg__, sum, mean, norm, square, __getitem__, __setitem__, check_format
+    :members: shape, context, dtype, stype, data, indices, indptr, copy, copyto, as_in_context, asscipy, asnumpy, asscalar, astype, tostype, slice, wait_to_read, zeros_like, round, rint, fix, floor, ceil, trunc, sin, tan, arcsin, arctan, degrees, radians, sinh, tanh, arcsinh, arctanh, expm1, log1p, sqrt, square, __neg__, sum, mean, norm, square, __getitem__, __setitem__, check_format, clip, sign
 
 .. autoclass:: mxnet.ndarray.sparse.RowSparseNDArray
     :members: shape, context, dtype, stype, data, indices, copy, copyto, as_in_context, asnumpy, asscalar, astype, tostype, wait_to_read, zeros_like, round, rint, fix, floor, ceil, trunc, sin, tan, arcsin, arctan, degrees, radians, sinh, tanh, arcsinh, arctanh, expm1, log1p, sqrt, square, __negative__, norm, __getitem__, __setitem__, check_format, retain, clip, sign
```
Comment (eric-haibin-lin) [inline docs/api/python/ndarray/sparse.md]: Also add abs for RowSparse: members?

Comment (haojin2) [reply docs/api/python/ndarray/sparse.md]: Added already: https://github.com/apache/incubator-mxnet/pull/11559/files#diff-3b14682dd131ed6386b9e343b0c707ebR407

Comment (haojin2) [reply docs/api/python/ndarray/sparse.md]: Oh you mean this one, will do soon.

Comment (haojin2) [reply docs/api/python/ndarray/sparse.md]: Done.

**Pull request merged**